testing/firestore/cmd/generate-firestore-tests/generate-firestore-tests.go

```diff
@@ -239,6 +239,51 @@ update operation should be produced.`,
 			maskForUpdate: nil,
 			transform:     transforms(st("a")),
 		},
+		{
+			suffix: "nested-single-value",
+			desc: 	`Updating a nested value results in update masks that are tightly 
+scoped to that specific field.`,
+			comment:   `Changing a.b sends an update that's scoped specifically to
+a.b, instead of sending an update that changes the entirety of a. For example,
+"its field_key should be a.b: 7, not a: b: 7 (which would entirely replace all of
+"a and blow away anything other than a.b).`,
+
+			// inData => json_data (non-paths) => ???
+			inData:        `{"a.b": 7}`,
+
+			// paths => field_paths (paths) => []firestore.Update.Path
+			paths:         [][]string{{"a", "b"}},
+			// inData => json_values => []firestore.Update.Val
+			values:        []string{`7`},
+
+			// outData => request.writes => request expectation
+			outData:       mp("a", mp("b", 7)),
+			// maskForUpdate => request.update_mask => request expectation
+			maskForUpdate: []string{"a.b"},
+			// unnecessary?
+			transform:     nil,
+		},
+		{
+			suffix: "nested-single-value-with-transform",
+			desc: 	`TODO.`,
+			comment:   `TODO.`,
+
+			// inData => json_data (non-paths) => ???
+			inData:        `{"a.b": "ServerTimestamp"}`,
```

Comment:
I'd prefer that this testcase mixed both a server timestamp *and* a normal subkey, e.g.:
```suggestion
			inData:        `{"a.b": 7, "a.c": "ServerTimestamp"}`,
```

Reply:
@tseaver Thanks for your comments. This is WIP. I'll ping when it's ready for re-review :) Also, I'm trying to reach you on hangouts if you're available?